src/components/Enemy.ts

```diff
@@ -61,7 +61,7 @@ export default class Enemy extends Phaser.Sprite {
 
         if ((this.last_fired + this.enemy_reload_time < now) && (this.health > 0)) {
             let bullet = this.bullets.getFirstExists(false);
-            bullet.reset(this.turret.x, this.turret.y);
+            bullet && bullet.reset(this.turret.x, this.turret.y);
```

Comment:
If bullet can be undefined, won't the next line fail also fail ?

Reply:
yes, good finding, i'll fix this